rust/private/rustdoc_test.bzl

```diff
@@ -24,10 +24,7 @@ def _rust_doc_test_impl(ctx):
 
     toolchain = find_toolchain(ctx)
 
-    dep_info = collect_deps(
```

Comment:
This is building a crate that has 1 direct dependency, so it's slightly different. 
Would prefer to revert this and the link_flags change, assuming that d.transitive_crates change was the actual fix.

Reply:
IIUC this is not true, rust doc test build integration test inside the crate. Anyway, just having transitive_crates is not enough, you actually need to define --extern for all direct deps of the dep of rust_doc_test.

Reply:
I think that logically each doctest is built in a separate crate, hence examples containing externs
https://github.com/bazelbuild/rules_rust/blob/78fc36d22e69a5113af63d59210a28c7ea55fbb8/tools/runfiles/runfiles.rs#L16-L20
Though that doesn't mean flags et al would be the same, as you've found..

(mostly for me:)
Cargo _appears_ to do the same thing, https://github.com/rust-lang/cargo/blob/18c8c5fb9934eefb554f915bccceada54c9c29c9/src/cargo/ops/cargo_test.rs#L181
where doctest.deps comes from https://github.com/rust-lang/cargo/blob/88839c97a544b9b87f9b48ad7e96233c5f35b99d/src/cargo/core/compiler/context/mod.rs#L230 but without L67